crates/macro/src/lib.rs

```diff
@@ -292,7 +292,12 @@ fn generate_code(
 
     let default_locale = if let Some(default_locale) = args.default_locale {
         quote! {
-            rust_i18n::set_locale(#default_locale);
+            use std::ops::Deref;
+            if #default_locale != rust_i18n::locale().deref() {
+                rust_i18n::set_locale(rust_i18n::locale().deref());
```

Comment:
There is not need to set again, just do nothing?

Reply:
Yes, I have tested it. This will cause it to get the value of ```CURRENT_LOCALE``` when it is init

Reply:
I mean this line set_locale is meaningless, the locale is already that value.

Reply:
The default value of ```CURRENT_LOCALE``` is "en". These codes will not read the value set by set_locale during ```runtime```. Instead, they will overwrite the set_locale of ```runtime``` to "en" during initialization. The purpose of this modification is to obtain and apply the modified ```CURRENT_LOCALE``` during runtime during initialization.